compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -33,6 +33,12 @@ use crate::errors::{
 
 mod invariant;
 
+/// The `ValTree` alongside its type.
+struct ValTreeAndTy<'tcx> {
+    valtree: ty::ValTree<'tcx>,
+    ty: Ty<'tcx>,
+}
```

Comment:
cc @BoxyUwU this is a practical type we could use in more places, and implement methods like `try_to_bits` directly on, to avoid having to pass the type separately

Reply:
Ah yeah sorry this was intended to be follow up work from the original PR that removed `Ty` from `ty::Const` but it just didn't wind up happening :+1:

Reply:
I have a branch that implements this PR's approach more globally (https://github.com/rust-lang/rust/compare/master...lukas-code:rust:typed-valtree), will send PR soon.

Reply:
With @lukas-code we talked about zulip. We thought we would remove this PR and keep his that will be coming in a moment.

I will look at that PR as soon as possible.